test/e2e/webhook_test.go

```diff
@@ -0,0 +1,82 @@
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
+
+)
+
+var _ = Describe("bundle api validating webhook", func() {
+	When("Bundle is valid", func() {
+		var (
+			bundle *rukpakv1alpha1.Bundle
+			ctx    context.Context
+		)
+		AfterEach(func() {
+			By("deleting the testing Bundle resource")
+			err := c.Delete(ctx, bundle)
+			Expect(err).To(BeNil())
+		})
+
+		It("should create the bundle resource", func() {
+			ctx = context.Background()
+
+			bundle = &rukpakv1alpha1.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "valid-bundle-",
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					ProvisionerClassName: plainProvisionerID,
+					Source: rukpakv1alpha1.BundleSource{
+						Type: rukpakv1alpha1.SourceTypeImage,
+						Image: &rukpakv1alpha1.ImageSource{
+							Ref: "testdata/bundles/plain-v0:valid",
+						},
+					},
+				},
+			}
+			err := c.Create(ctx, bundle)
+			Expect(err).To(BeNil())
+		})
+	})
+
+	When("Bundle name is too long", func() {
+		var (
+			bundle *rukpakv1alpha1.Bundle
+			ctx    context.Context
+		)
+		AfterEach(func() {
+			By("deleting the testing Bundle resource for failure case")
+			err := c.Delete(ctx, bundle)
+			Expect(err).NotTo(BeNil())
+		})
+
+		It("should fail the bundle creation", func() {
+			ctx = context.Background()
+
+			bundle = &rukpakv1alpha1.Bundle{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "bundlename-0123456789012345678901234567891",
+				},
+				Spec: rukpakv1alpha1.BundleSpec{
+					ProvisionerClassName: plainProvisionerID,
+					Source: rukpakv1alpha1.BundleSource{
+						Type: rukpakv1alpha1.SourceTypeImage,
+						Image: &rukpakv1alpha1.ImageSource{
+							Ref: "testdata/bundles/plain-v0:valid",
+						},
+					},
+				},
+			}
+			err := c.Create(ctx, bundle)
+			Expect(err).NotTo(BeNil())
```

Comment:
Is there any way we can perform this logic in a BeforeEach clause, and the It spec compares the error with the ContainSubstring gomega matcher?

Reply:
Alternatively: we just deflate this test case entirely as a top-level It spec if we don't have the need for more test cases right now?

Reply:
Another thing that comes to mind: should we be testing this workflow from the kubectl [create,apply] -f perspective? Is there any potential for skew between using the dynamic client, and a local kubectl CLI experience?

Reply:
Yes, I'll try that.  Thanks!

Reply:
OK.  I'll just restructure the test.  Thanks!

Reply:
> Is there any way we can perform this logic in a BeforeEach clause, and the It spec compares the error with the ContainSubstring gomega matcher?

+1 from me on this suggestion. The `BeforeEach` setup should match the text of what the parent `When` is describing.

Reply:
OK.  I'll change to put it into BeforeEach